pipeline/authn/authenticator_cookie_session.go

```diff
@@ -155,6 +155,10 @@ func forwardRequestToSessionStore(r *http.Request, checkSessionURL string, prese
 
 	// We need to make a COPY of the header, not modify r.Header!
 	for k, v := range r.Header {
+		// remove websocket-related headers to not confuse reverse proxies
+		if k == "Upgrade" || k == "Connection" || k == "Sec-Websocket-Key" || k == "Sec-Websocket-Version" {
```

Comment:
One problem I realized just now is that the header key might not be canonicalized, meaning that it could also start with a lower case so e.g. `upgrade` or `uPGRADE` depending on how buggy the client is. If you change this to:

```
switch strings.ToLower(k) {
case "upgrade":
  ...
}
```

this should no longer be an issue!

Reply:
Go should canonicalize these at request parse time, if I read this right: https://cs.opensource.google/go/go/+/refs/tags/go1.18.1:src/net/http/request.go;l=138-160

Reply:
👍 agreed, I'll try to figure out a test that actually tests this (I think `Header.Set` also canonicalizes 🤪)